lib/pluggy_elixir/error.ex

```diff
@@ -0,0 +1,19 @@
+defmodule PluggyElixir.Error do
+  @moduledoc """
+  Parse API error responses.
+  """
+  alias PluggyElixir.Error.Unauthorized
+  alias PluggyElixir.HttpAdapter.Response
+
+  @spec parse(Response.t()) :: Unauthorized.t()
+
+  @doc """
+  Return an Error struct by given response
+  """
+  def parse(%Response{} = response) do
```

Comment:
```suggestion
  @spec parse(Response.t()) :: Unauthorized.t()
  def parse(%Response{} = response) do
```

What do you think about bringing the `spec` closer to the function itself?

Reply:
An example from `Ecto.Changeset`:

https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/changeset.ex#L375

Reply:
👍🏻